ffmuc-mesh-vpn-wireguard-vxlan/files/lib/gluon/gluon-mesh-wireguard-vxlan/checkuplink

```diff
@@ -18,24 +18,42 @@ clean_port() {
 	echo "$(echo $1 | sed -r 's/:[0-9]+$|\[|\]//g')"
 }
 
-check_address_family() {
-	local peer_publickey="$1"
-	local peer_endpoint="$2"
-	local gateway="$(clean_port $peer_endpoint)"
+extract_port() {
+    local address_port="$1"
+
+    # Remove brackets for IPv6 addresses
+    local address="$(echo "$address_port" | sed -e 's/\[//;s/\]//')"
+
+    # Extract port number
+    echo "$address" | awk -F: '{print $NF}'
+}
+
+combine_ip_port() {
+    local ip="$1"
+    local port="$2"
+
+    if echo "$ip" | grep -q ":"; then
+        echo "[$ip]:$port"
+    else
+        echo "$ip:$port"
+    fi
```

Comment:
You probably want to use regex matching instead and maybe simplify the if/else

```suggestion
    if [[ ${ip} == *":"* ]]; then
            ip="[${ip}]"
    fi
    echo "$ip:$port"
```

Reply:
Something wrong happened here during rebase. The function definition of `combine_ip_port` disappeared

Reply:
Whoops, readded, and applied your change suggestion

Reply:
Hm, shellcheck says
```
  ./ffmuc-mesh-vpn-wireguard-vxlan/files/lib/gluon/gluon-mesh-wireguard-vxlan/checkuplink:39:8: error: In dash, [[ ]] is not supported. (ShellCheck.SC3010)
```
Is this true? It does seem to work on a node, tough...